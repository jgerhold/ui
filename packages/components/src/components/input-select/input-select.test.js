import { mount } from '@vue/test-utils'
import InputSelect from './InputSelect'

describe('InputSelect', () => {
  describe('component', () => {
    test('should be a Vue instance', () => {
      const wrapper = mount(InputSelect)
      expect(wrapper.isVueInstance()).toBeTruthy()
    })
  })

  describe('props', () => {
    test('should render with default props', () => {
      const wrapper = mount(InputSelect)
      expect(wrapper.element).toMatchSnapshot()
    })

    const props = [
      {
        prop: 'disabled',
        value: true
      },
      {
        prop: 'color',
        value: '#fff'
      },
      {
        prop: 'background',
        value: '#fff'
      },
      {
        prop: 'borderColor',
        value: '#fff'
      },
      {
        prop: 'value',
        value: 'foo'
      }
    ]

    props.forEach(({ prop, value }) => {
      test(`should render type '${prop}'`, async () => {
        const wrapper = await mount(InputSelect, {
          propsData: {
            [prop]: value,
            options: ['foo', 'bar', 'baz']
          }
        })

        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })

  describe('events', () => {
    test('should dispatch change event on change', async () => {
      const wrapper = await mount(InputSelect, {
        propsData: {
          options: ['foo', 'bar', 'baz']
        }
      })

      wrapper
        .findAll('option')
        .at(1)
        .trigger('change')
      expect(wrapper.emitted('change')).toEqual([['bar']])
    })
  })
})
