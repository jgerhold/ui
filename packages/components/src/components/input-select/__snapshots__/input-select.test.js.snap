// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSelect props should render type 'background' 1`] = `
<select
  class="input-select"
  style="color: rgb(43, 138, 198); background: rgb(255, 255, 255); border-color: #2b8ac6;"
>
  <option>
    
    foo
  
  </option>
  <option>
    
    bar
  
  </option>
  <option>
    
    baz
  
  </option>
</select>
`;

exports[`InputSelect props should render type 'borderColor' 1`] = `
<select
  class="input-select"
  style="color: rgb(43, 138, 198); background: rgb(255, 255, 255); border-color: #fff;"
>
  <option>
    
    foo
  
  </option>
  <option>
    
    bar
  
  </option>
  <option>
    
    baz
  
  </option>
</select>
`;

exports[`InputSelect props should render type 'color' 1`] = `
<select
  class="input-select"
  style="color: rgb(255, 255, 255); background: rgb(255, 255, 255); border-color: #2b8ac6;"
>
  <option>
    
    foo
  
  </option>
  <option>
    
    bar
  
  </option>
  <option>
    
    baz
  
  </option>
</select>
`;

exports[`InputSelect props should render type 'disabled' 1`] = `
<select
  class="input-select"
  disabled="disabled"
  style="color: rgb(43, 138, 198); background: rgb(255, 255, 255); border-color: #2b8ac6;"
>
  <option>
    
    foo
  
  </option>
  <option>
    
    bar
  
  </option>
  <option>
    
    baz
  
  </option>
</select>
`;

exports[`InputSelect props should render type 'value' 1`] = `
<select
  class="input-select"
  style="color: rgb(43, 138, 198); background: rgb(255, 255, 255); border-color: #2b8ac6;"
>
  <option>
    
    foo
  
  </option>
  <option>
    
    bar
  
  </option>
  <option>
    
    baz
  
  </option>
</select>
`;

exports[`InputSelect props should render with default props 1`] = `
<select
  class="input-select"
  style="color: rgb(43, 138, 198); background: rgb(255, 255, 255); border-color: #2b8ac6;"
/>
`;
